experimental/ComplexReflectionGroups/src/unitary_matrices.jl

```diff
@@ -0,0 +1,51 @@
+export is_unitary
+
+# Unitary matrices
+#
+# Ulrich Thiel, 2023
+
+function is_unitary(M::AbstractAlgebra.Generic.MatSpaceElem{AbsSimpleNumFieldElem})
+
+    if !is_square(M)
+        return false
+    end
+
+    # create the conjugate transpose of M
+    K = base_ring(M)
+    n = ncols(M)
+    conj = complex_conjugation(K)
+    Mct = transpose(matrix(K, n, n, [conj(x) for x in M]))
+
+    return M*Mct == identity_matrix(K,n)
+
+end
+
+function is_unitary(M::AbstractAlgebra.Generic.MatSpaceElem{T}) where T <: Union{Hecke.RelSimpleNumFieldElem, AbsNonSimpleNumFieldElem}
+
+    if !is_square(M)
+        return false
+    end
+
+    # convert base field to absolute and convert matrix as well
+    K = base_ring(M)
+    n = ncols(M)
+    L, f = absolute_simple_field(K)
+    g = inv(f)
+    ML = matrix(L, n, n, [g(x) for x in M])
```

Comment:
Hm, does this also work?
```suggestion
    ML = map_entries(g, M)
```